### ISSUE TITLE
[OJS] Improve submission process

### ISSUE THREAD

**Issue by NateWr** [pkp/pkp-lib#1528]
This issue covers improvements to be made to the submission process. This includes improving how users discover and initiate an attempt to submit, the registration process, and the actual submission flow.

An initial comment: https://github.com/pkp/pkp-lib/issues/975#issuecomment-166318343

> The initial submission flow is disorienting because users have to first learn that they have to register and then once registered they get dumped in a very different-looking backend display.
> ...
> Create a frontend landing page for a completed registration. The page should say prominently "Thanks for registering with the Journal/Press of ..." and followed by a list of actions: "Make a Submission", "Complete Profile". Only then should it jump to the backend. This landing page could be used for logging in too, and include other actions based on user role, eg: "View assigned submissions" for reviewers.

In addition, a review of the submission process threw up a number of smaller improvements that could be made.

The following is a complete list of tasks intended for completion for OJS 3.0:
- [x] Display more prominent submissions button on the frontend. https://github.com/pkp/pkp-lib/issues/1717
- [x] Create a login/registration flow which takes the user directly from login/registration to the submission workflow, or provides a landing page after login/registration that is native to the frontend and provides clear options for initiating a submission.
- [x] Tidy up the copyright notice display at the top of the initial submission page
- [x] During file upload there's a message to contact admin. Remove it. https://github.com/pkp/pkp-lib/issues/1528#issuecomment-227197491
- [x] Add contributor: "Denotes required field" indicator appears below the buttons
- [x] Add contributor: asks for URL, ORCID, Suffix, Affiliation, Bio. ~~Do contributors get accounts and can they be asked to enter this detail later?~~ Should be kept but could maybe be tidied up a bit. https://github.com/pkp/pkp-lib/issues/1528#issuecomment-227197491
- [ ] Confirmation: actually show a summary of their submission details. The more the merrier.
- [x] ~~Complete: maybe a little style love for the "next steps" links.~~~
- [x] ~~~Complete: should we add a link to the user's profile?~~~
- [x] Copy Orcid ID from submission user to main author. https://github.com/pkp/pkp-lib/issues/1528#issuecomment-230051238
- [ ] Improve clarity of file type selection. https://github.com/pkp/pkp-lib/issues/1528#issuecomment-307388948

**Comment by asmecher** [comment]
> During file upload there's a message to contact admin. Is this valuable or could it be removed?

This is probably just tradition. I don't suspect there's much utility in it.

> Do contributors get accounts and can they be asked to enter this detail later?

Contributors do not get accounts automatically -- they'll have to register separately. Author records are maintained separately from user accounts, so ORCiDs will eventually be supported as a linking aid. Currently for other authors to be granted access they'll need to have a user account assigned through the participants grid, and this is done separately of the author list that's entered in the submission process for metadata purposes.

**Comment by NateWr** [comment]
Thanks @asmecher!

**Comment by crism** [comment]
ORCID should be copied (along with affiliation and nation) from submitting user to the main author; currently, it’s blank for the author, even if the user has set it in their profile. (See #1450.)

**Comment by NateWr** [comment]
I've implemented several of the key parts of this issue. When these PRs are merged this can be reassigned to OJS 3.0.1.

PRs:
https://github.com/pkp/pkp-lib/pull/1716
https://github.com/pkp/ojs/pull/968
https://github.com/pkp/omp/pull/320

**Comment by asmecher** [comment]
Don't forget to assign to a reviewer per the rota :)

**Comment by NateWr** [comment]
Just waiting on the tests to fail. :)

**Comment by NateWr** [comment]
Um... apparently the tests can't find the logout link now?

https://travis-ci.org/pkp/omp/jobs/151269634#L834
https://travis-ci.org/pkp/omp/jobs/151269633#L829

**Comment by asmecher** [comment]
When the registration page finishes, the user has a session but it isn't associated with their user ID. Hence, when generating a page, no "Log Out" link. The redirect that used to be there at the end of registration would cause a full page reload, and on that page "Log Out" would be present.

**Comment by NateWr** [comment]
Hmm, the OJS tests seemed to pass:

https://travis-ci.org/pkp/ojs/builds/151269668

I wonder if the logout issue is unique to OMP because of https://github.com/pkp/pkp-lib/issues/1715.

**Comment by asmecher** [comment]
@NateWr, have you walked through the registration process to see whether the Log Out link is presented in the right place? Or does this seem to be just a test issue?

**Comment by NateWr** [comment]
You were right. :) I've solved this by redirecting to a new `/user/registrationComplete` page on the frontend. Tests are running...

**Comment by NateWr** [comment]
@asmecher The tests are written assuming that registration dumps a user in the backend Submissions page, and navigating to a new submission is built into the `createSubmission` method. A couple options, which would you prefer:

1. I write a new test method for navigating from registration to a new submission, and modify the `createSubmission` method to start from the first step of the submission page.
2. Just modify `createSubmission` to assume it's starting from the registration page. A quick glance suggests it's only used immediately following a registration, but future tests might want to use it differently...

**Comment by asmecher** [comment]
#2 is OK, at least for now.

**Comment by NateWr** [comment]
The OJS tests are passing but OMP is failing, on code that should be identical. I believe this is happening because of #1715, but that should mean that all tests in OMP are failing. Maybe #1715 needs to be sorted out sooner rather than later?

**Comment by asmecher** [comment]
@NateWr, see https://github.com/pkp/pkp-lib/issues/1715#issuecomment-239836280

**Comment by NateWr** [comment]
I think this one fell off the train somewhere. I'll rebase and re-run tests tomorrow.

**Comment by NateWr** [comment]
I rebased these today for merge, but the OMP test is failing now. It appears to time out. I restarted it a couple times but it keeps timing out:

https://travis-ci.org/pkp/omp/jobs/154720368

Any thoughts?

**Comment by asmecher** [comment]
Are the PRs updated with your rebase? The OMP PR still looks like it has a merge conflict (can't be merged).

**Comment by NateWr** [comment]
Hmm, I re-re-based but now the tests are wigging out. The OJS is giving lots of output I've never seen before. The OMP test is still failing on the "Make a New Submission" string. I've manually walked through a registration and the "Make a New Submission" string is present.

Not sure what to do with this one...

**Comment by NateWr** [comment]
@asmecher Sorry, I'm running out of ideas on this one too.

The OJS test was passing, I've just rebased to fix a merge conflict so they're re-running.
The OMP tests for `mysql` and `pgsql` are failing at different places, but I can't see any particular reason for it.

**Comment by asmecher** [comment]
I see the tests are re-running here too -- I'll leave tabs open for when they finish. I may have to disappear for a while before that happens.

**Comment by asmecher** [comment]
Tests passed!

**Comment by NateWr** [comment]
@asmecher 

Same here as #1739. OJS tests are passing but OMP tests are failing on `AclarkSubmissionTest`.

**Comment by NateWr** [comment]
PRs merged. Rescheduling the reset for post-3.0.

**Comment by ctgraham** [comment]
The `./user/registrationComplete` introduced here is:

- Accessible outside of the workflow of registration
- Accessible following registration at a site context, but displaying a "make a new submission" link. (This introduces a "No _[context]_ in context" error if followed. (cf. http://forum.pkp.sfu.ca/t/modifying-userregistercomplete-tpl/30720/1)

Should this be followed up here, or introduced as a new issue?

**Comment by NateWr** [comment]
Consolidating a related comment from a separate issue https://github.com/pkp/pkp-lib/issues/975#issuecomment-164777350:

> We could make primary or common file component types easier to select. The list is long and sorted alphabetically so you have to go digging for the "main file".
>
> ![submission-component-list](https://cloud.githubusercontent.com/assets/2306629/11812895/779ef7cc-a336-11e5-9c69-3b3beaf90fd6.png)

**Comment by NateWr** [comment]
There are only three items left, and I don't think any will make it into 3.1. The only exception is maybe:

"Copy Orcid ID from submission user to main author https://github.com/pkp/pkp-lib/issues/1528#issuecomment-230051238" 

@asmecher I'm going to bump this to OJS 3.2 unless you want to split off the ORCID task for OJS 3.1.

**Comment by asmecher** [comment]
The ORCID change was trivial; I've just committed it.

**Comment by NateWr** [comment]
The remaining items in the checklist at the top are covered by #7191
